client/network/src/config.rs

```diff
@@ -643,6 +648,115 @@ impl NonReservedPeerMode {
 	}
 }
 
+///Doc
+#[derive(Clone, Debug)]
+pub enum NodePreShareKeyConfig {
+	/// A Ed25519 secret key configuration.
+	PRESHAREDKEY(PreSharedKeySecret),
+}
+
+/// The options for obtaining a Ed25519 secret key.
+pub type PSKey = PreSharedKeySecret;
+
+///Doc
+#[derive(Clone)]
+pub enum PreSharedKeySecret {
+	///Doc
+	Rpc(std::net::SocketAddr),
+	///Doc
+	File(PathBuf),
+	///Doc
+	Hard,
+}
+
+impl Default for NodePreShareKeyConfig {
+	fn default() -> NodePreShareKeyConfig {
+		Self::PRESHAREDKEY(PreSharedKeySecret::Hard)
+	}
+}
+
+impl fmt::Debug for PreSharedKeySecret {
```

Comment:
We will always read this key from the file. If file is not present, then we call RPCs from other nodes to obtain it and put to this file.